### PR TITLE
Cleanup, Vue CLI, Shaka

### PR DESCRIPTION
**Pull request by ttshivers**
- Simplified room creation and joining process
- Better web player (same as what Plex web uses)
	- Direct play and direct stream support
	- Native subtitle support
	- Skip next / previous controls to easily navigate episodes in a show
	- Better high bitrate support with no more "appendBuffer" errors
- Skip intro / auto skip intro
- Auto Host (Anyone can start playing new content rather than just the host)
- Sound / desktop notifications for chat
- Major rewrite / cleanup of sync logic

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,106 +1,78 @@
...
-    "videojs-contrib-hls": "^5.15.0",
-    "vue": "^2.6.10",
-    "vue-cookies": "^1.7.0",
-    "vue-router": "^2.3.1",
-    "vuetify": "^1.5.14",
-    "vuex": "^2.3.1",
-    "waterline": "^0.13.6",
-    "waterline-mysql": "^0.6.0"
+    "nconf": "^0.10.0",
+    "syncloungesocket": "^5.3.4"
```
Comment (gcordalis) [inline package.json:21]: Need to take a look at syncloungesocket

Comment (ttshivers) [reply package.json:21]: It can be found https://github.com/ttshivers/syncloungesocket

File: src/store/modules/slplayer/actions.js [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+    } else {
+      await dispatch('DESTROY_ASS');
+    }
+  },
+
+  CHANGE_PLAYER_SRC: async ({ getters, commit, dispatch }) => {
+    console.debug('CHANGE_PLAYER_SRC');
+
+    // Abort subtitle requests now or else we get ugly errors from the server closing it.
+    await dispatch('DESTROY_ASS');
```
Comment (samcm) [inline src/store/modules/slplayer/actions.js:91]: lmao

**Comment by ttshivers** [comment]
Replaced by #229 to merge in default branch

**Pull request closed without merging**